.docker/build/build.sh

```diff
@@ -169,6 +169,11 @@ function refresh_hints() {
     # Add all the hints, even those not under version control
     find $hints_dir -iname '*.hints' -and -not -path '*/.*' -and -not -path '*/dependencies/*' | xargs git add
 
+    # Add new files from the C snapshot in dist.
+    # Remove files first that we don't want in the snapshot.
+    find dist \( -name "*.d" -o -name "*.cmi" -o -name "*.cmx" -o -name "*.ocaml" -o -name "*.cmxa" \) | xargs rm
+    git add dist
```

Comment:
In FStar, the OCaml snapshot is refreshed by passing `"git ls-files src/ocaml-output/ | xargs git add"` as the second argument to `refresh_hints` (i.e. `$extra`). That is, we only refresh snapshot files that are already under version control. I suggest we do the same for uniformity and to be aware of newly generated files--the person whose change causes the new file to be generated will need to check it in explicitly.

On the other hand, snapshots are less stable, so we may want to automatically add untracked files as well. In this case, we need to add `*.dll` the toplevel `.gitignore`. The other patterns (e.g. `*.d`, `*.cmi`, etc.) are already ignored.

Reply:
I'm fine with both. Really depends on how stable the snapshots are supposed to be I think. 

Maybe we can start with only refreshing existing files as that's less controversial and change it later if people want to automate adding new files?

Reply:
Yes, that's a good idea. We can discuss this on the Hacl meeting today.

To be clear, that would mean removing the lines above and passing
`"git ls-files dist | xargs git add"` instead of `true` to `refresh_hints_and_dist`